jetbrains-core/src/software/aws/toolkits/jetbrains/services/clouddebug/actions/InstrumentDialogWrapper.kt

```diff
@@ -56,9 +59,23 @@ class ConfirmNonProductionDialogWrapper(project: Project, serviceName: String) :
     override fun createSouthAdditionalPanel() = JPanel(GridBagLayout()).apply { add(doNotShowAgain) }
 
     override fun doOKAction() {
+        recordTelemetry(TelemetryConstants.TelemetryResult.Succeeded)
         if (doNotShowAgain.isSelected) {
             settings.showEnableDebugWarning = false
         }
         super.doOKAction()
     }
+
+    override fun doCancelAction() {
+        recordTelemetry(TelemetryConstants.TelemetryResult.Cancelled)
+        super.doCancelAction()
+    }
+
+    private fun recordTelemetry(result: TelemetryConstants.TelemetryResult) {
+        telemetry.record(project) {
+            datum("${TelemetryConstants.CLOUDDEBUG_TELEMETRY_PREFIX}_confirm_enable") {
+                metadata(TelemetryConstants.RESULT, result.name)
```

Comment:
add an explicit call to `count()` since it's only implicit in the action wrapper system